### PR TITLE
team crudlike structre, mapstruct dependency

### PR DESCRIPTION
**Pull request by macuzarski**
had to add dependencies to POM, unfortunately there seems to be some issue with just toList() method on streams with MapStruct, so I had to convert it to using Collectors.toList() in UserServiceImp, it shouldn't be an issue to change it for the production branch

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -167,6 +167,11 @@
 			<groupId>org.mapstruct</groupId>
 			<artifactId>mapstruct</artifactId>
 			<version>1.5.3.Final</version>
+    </dependency>
+    <dependency>
```
Comment (marcinkunert) [inline pom.xml:171]: formatting seems to be off

File: pom.xml [hunk 2]
```diff
@@ -180,6 +185,29 @@
 					<path>/</path>
 				</configuration>
 			</plugin>
+			<plugin>
+				<groupId>org.apache.maven.plugins</groupId>
+				<artifactId>maven-compiler-plugin</artifactId>
+				<version>3.5.1</version>
+				<configuration>
+					<source>1.8</source>
+					<target>1.8</target>
```
Comment (marcinkunert) [inline pom.xml:194]: You should not override the properties:
```
		<maven.compiler.source>17</maven.compiler.source>
		<maven.compiler.target>17</maven.compiler.target>
```

File: src/main/java/com/softserve/sportshub/team/Team.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @Column
+    private String teamName;
+
+    //TODO
+//    @Column
+//    private Category category;
+//    @Column
+//    private SubCategory subCategory;
```
Comment (marcinkunert) [inline src/main/java/com/softserve/sportshub/team/Team.java:25]: Do not put unfinished work to be merged to main

File: src/main/java/com/softserve/sportshub/team/TeamServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    @Override
+    public List<Team> findAll() {
+        return teamDao.findAll();
+    }
+
+    //TODO
+    @Override
+    public Team findByName(String name) {
+        return null;
+    }
```
Comment (marcinkunert) [inline src/main/java/com/softserve/sportshub/team/TeamServiceImpl.java:23]: Do not put unfinished work to be merged to main

File: src/main/resources/hibernate.cfg.xml [hunk 5]
```diff
@@ -9,7 +9,7 @@
         <property name="hibernate.show_sql">true</property>
         <property name="hibernate.connection.driver_class">com.mysql.cj.jdbc.Driver</property>
         <property name="hibernate.connection.username">root</property>
-        <property name="hibernate.connection.password">admin123</property>
+        <property name="hibernate.connection.password">password</property>
```
Comment (marcinkunert) [inline src/main/resources/hibernate.cfg.xml:12]: was this done on purpose?

File: src/main/java/com/softserve/sportshub/team/TeamController.java [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+    private TeamService teamService;
+
+    @Autowired
+    private TeamTeamDtoMapper mapper;
+
+    @GetMapping("/")
+    public List<TeamDto> getTeams() {
+        List<Team> teams = teamService.findAll();
+
+        return mapper.teamListToTeamListDto(teams);
```
Comment (marcinkunert) [inline src/main/java/com/softserve/sportshub/team/TeamController.java:24]: You most likely shouldn't leak entities into controllers. Services should return dtos

**Review by kmpol — APPROVED** [review]

**Review by PawelMiskowiec — APPROVED** [review]

**Review by remkro — APPROVED** [review]

**Pull request merged**